workflow/djworker.py

```diff
@@ -0,0 +1,150 @@
+import datajoint as dj
+import inspect
+import time
+import os
+import platform
+from datetime import datetime
+
+
+_populate_settings = {
+    'display_progress': True,
+    'reserve_jobs': True,
+    'suppress_errors': True}
+
+
+class WorkerLog(dj.Table):
+    definition = """
+    # Registration of processing jobs running .populate() jobs (e.g. in process.py script)
+    process_timestamp : datetime      # timestamp of the processing job
+    process           : varchar(64)
+    ---
+    worker_name=''    : varchar(255)  # name of the worker
+    host              : varchar(255)  # system hostname
+    user=''           : varchar(255)  # database user
+    pid=0             : int unsigned  # system process id  
+    """
+
+    _table_name = '~WorkerLog'
+
+    @classmethod
+    def log_process_job(cls, process, worker_name='', db_prefix=''):
+        if isinstance(process, dj.user_tables.TableMeta):
+            schema_name, table_name = process.full_table_name.split('.')
+            schema_name = schema_name.strip('`').replace(db_prefix, '')
+            table_name = dj.utils.to_camel_case(table_name.strip('`'))
+            process_name = f'{schema_name}.{table_name}'
+            user = process.connection.get_user()
+        elif inspect.isfunction(process):
+            process_name = process.__name__
+            user = ''
+
+        if not worker_name:
+            frame = inspect.currentframe()
+            function_name = frame.f_back.f_code.co_name
+            module_name = inspect.getmodule(frame.f_back).__name__
+            worker_name = f'{module_name}.{function_name}'
+
+        cls.insert1({'process': process_name,
+                     'process_timestamp': datetime.utcnow(),
+                     'worker_name': worker_name,
+                     'host': platform.node(),
+                     'user': user,
+                     'pid': os.getpid()})
+
+    @classmethod
+    def print_recent_jobs(cls, backtrack_minutes=60):
+        recent = (cls.proj(
+            minute_elapsed='TIMESTAMPDIFF(MINUTE, process_timestamp, UTC_TIMESTAMP())')
+                  & f'minute_elapsed < {backtrack_minutes}')
+
+        recent_jobs = dj.U('process').aggr(
+            cls & recent,
+            worker_count='count(DISTINCT pid)',
+            minutes_since_oldest='TIMESTAMPDIFF(MINUTE, MIN(process_timestamp), UTC_TIMESTAMP())',
+            minutes_since_newest='TIMESTAMPDIFF(MINUTE, MAX(process_timestamp), UTC_TIMESTAMP())')
+
+        return recent_jobs
+
+    @classmethod
+    def delete_old_logs(cls, cutoff_days=3):
+        old_jobs = (cls.proj(
+            elapsed_days=f'TIMESTAMPDIFF(DAY, process_timestamp, "{datetime.utcnow()}")')
+                    & f'elapsed_days > {cutoff_days}')
+        if old_jobs:
+            with dj.config(safemode=False):
+                (cls & old_jobs).delete()
```

Comment:
consider using `delete_quick`
```suggestion
                (cls & old_jobs).delete_quick()
```